underscore.js

```diff
@@ -521,8 +521,9 @@
   };
 
   // Flatten out an array, either recursively (by default), or just one level.
-  _.flatten = function(array, shallow) {
-    return flatten(array, shallow, false);
+  _.flatten = function(array, depth) {
+    if (!depth) depth = Infinity;
```

Comment:
Should return a shallow clone/slice which should shake out naturally if `0` is allow to be used as a depth (no need for a `if` or special branch)